packages/hint-compat-api/src/helpers/compat-base.ts

```diff
@@ -0,0 +1,57 @@
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+
+import { CachedCompatFeatures } from './cached-compat-features';
+import { TestFeatureFunction, FeatureInfo, MDNTreeFilteredByBrowsers } from '../types';
+import { Events, Event } from 'hint/dist/src/lib/types';
+
+export abstract class CompatBase<T extends Events, K extends Event> {
+    protected testFunction: TestFeatureFunction;
+    protected hintContext: HintContext<T>;
+    protected hintResource: string = 'unknown';
+    protected MDNData: MDNTreeFilteredByBrowsers;
+    private cachedFeatures: CachedCompatFeatures;
+
+    public abstract async searchFeatures(parser: K): Promise<void>
+
+    public constructor(hintContext: HintContext<T>, MDNData: MDNTreeFilteredByBrowsers, testFunction: TestFeatureFunction) {
+        if (!testFunction) {
+            throw new Error('You must set a test function before testing a feature.');
+        }
+
+        this.cachedFeatures = new CachedCompatFeatures();
+        this.testFunction = testFunction;
+        this.hintContext = hintContext;
+        this.MDNData = MDNData;
+
+        // Clear feature cache between scans. Contexts like extension-vscode scan multiple times.
+        (this.hintContext as HintContext<Events>).on('scan::end', () => {
+            this.cachedFeatures = new CachedCompatFeatures();
+        });
```

Comment:
@borgitas21 I figured out why results were disappearing in the VS Code extension - the feature cache was never being cleared/reset. In some contexts we can run multiple scans with the same hint instances, so this state needed cleared to allow reporting issues again on subsequent scans. 

I went ahead and just made the fix, but feel free to refactor or move this if you have a different preference on how this should be done.